### PR TITLE
VACMS-6645: Create table audit view.

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Relates to #6645 . Other ACs will be covered in the original ticket. I have created a view that pulls all nodes with tables in field_content_block. I have separately compiled a list of ALL table instances through DB queries which can be found [here](https://airtable.com/apphETq1vi4mpYD0J/tblXbMw6Ylj6bkNcf/viwVI1TKs3SJm2uKb?blocks=bipnkHDa8xJbjId1c).

## Testing done
Searched for content that contains tables and saw them.

## Screenshots
image
image

## QA steps

As cms user
1. Visit the table audit (`/admin/content-tools/audit/table-audit`)
2. Then validate Acceptance Criteria from issue
  - [x] database queries (Drupal view?) for what nodes have tables, how many tables, what sections those nodes belong to
  - [x] display machine name should not be default
  - [x] show result count in header
  - [x] default sort is last update (newest at top)
  - [x] Reset button exists
  - [x] Apply button says 'filter'
  - [ ] default columns
    - [x] Title (sortable) 
    - [x] node ID
    - [x] Edit button  
    - [x] Content type (sortable)
    - [x] Moderation State
    - [x] Section (field_administration) (sortable)
    - [x] Updated (date time and by user) (sortable)
  - [ ] filters
    - [x] text search (exposed)
    - [x] content type (exposed)
    - [x] moderation state (exposed)
    - [x] section (exposed)
- [x] Filtering by major section (VISN1)  should include all items beneath that section including sub-sections.
  
  
  ### Definition of Done
  
  - [ ] Documentation has been updated, if applicable.
  - [ ] Tests have been added if necessary.
  - [ ] Automated tests have passed.
  - [ ] Code Quality Tests have passed.
  - [ ] Acceptance Criteria in related issue are met.
  - [ ] Manual Code Review Approved.
  
  ### Select Team for PR review
  
  - [ ] `Platform CMS Team`
  - [ ] `Sitewide CMS Team`
    - [x] `⭐️ Content ops`
    - [ ] `⭐️ CMS Experience`
    - [ ] `⭐️ Offices`
    - [ ] `⭐️ Product Support`
  
  
  ### Is this PR blocked by another PR?
  
  - [ ] `DO NOT MERGE`
  
  ### Does this PR need review from a Product Owner
  
  - [ ] `Needs PO review`
  
  ### CMS user-facing annoucement
  
  Is an announcement needed to let editors know of this change?
  - [ ] Yes, and it's written in issue ____ and queued for publication.
    - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
  - [ ] Yes, but it hasn't yet been written
    - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
  - [ ] No announcement is needed for this code change.
    - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
Given the great alpha sorting debacle of D9.3 this is going to need to be config imported and config exported again, other wise the next person to export will get an unanticipated surprise.

**Comment by swirtSJW** [comment]
Also the style of this View needs to be synced up with the style of our other Views.  Consistency of patterns is important for architecture.
image

**Comment by msbtterswrth** [comment]
Should be all set @swirtSJW, just waiting for tests now.

**Comment by swirtSJW** [comment]
> Fwiw, the only views I found with that style of revision (Updated) content were the newer facility views. I went ahead and updated this to match but there are many many other views that don't have that style. I can add it to the list of things to update in the other 'improve views' ticket.

Maybe I am missing something.  Here's at least 6 that have it
image

Many of the ones that don't have it were intentional deviations that needed the ability to sort updated separate from user.  Others were either old or I slipped up on catching it when they came in.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This is shaping up nicely.  Nice work on the table display in the table.

I am sorry, I missed a couple things on final review.  In general we want to mimic admin/content unless there is a good reason to deviate.
- [x] results counter (see admin/content)
- [x] reset button (see admin/content)
- [x] section filter is not the right section filter.  (see admin/content)

image

image

**Comment by msbtterswrth** [comment]
@swirtSJW good catch! This was such an old view, I made it before you showed me some things and I forgot it hadn't been updated!

**Review by swirtSJW — CHANGES_REQUESTED** [review]
image
@msbtterswrth  can you please add QA steps that you have run through so I that these kinds of things are tested and not left to chance of the reviewer.   The QA steps here presently only detect if the page exists.

**Comment by msbtterswrth** [comment]
@steve my bad, I must not have committed my last change. Just pushed that up now and also updated the ACs to be a bit more verbose.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
QA does not match QA step validation
Filter button text 
image

Edit button next to title.
image

This pattern does not reflect existing pattern.  
image

**Comment by msbtterswrth** [comment]
Updated. I realized also the ID column was missing from the ACs so that was built correctly just not documented correctly.

File: config/sync/views.view.table_audit.yml [hunk 1]
```diff
@@ -0,0 +1,1127 @@
...
+        result:
+          id: result
+          table: views
+          field: result
+          relationship: none
+          group_type: group
+          admin_label: ''
+          plugin_id: result
+          empty: false
+          content: "<p><a href=\"admin/content/audit/tables-usage-audit.csv\" class=\"button\">Download CSV</a></p>\r\n\r\n<p>Displaying @start - @end of @total</p>"
```
Comment (swirtSJW) [inline config/sync/views.view.table_audit.yml:1071]: This button does not work.
image


 I don't see the button in the ACs so maybe it is not needed.  We either need to fix it or remove it.
If we are keeping it, please add a QA step to test the functionality of this button.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
A couple things not working.  See code comments below.

This is why QA steps, not just ac's are important.

The bit about the edit button.  The majority of our other views have the edit button follow the title.  I do see we have one audit where it follows the id.  I think that was probably an oversight.   I am glad you started that doc because I think we want to prescribe the pattern as edit sits to the right of title.   I can't think of a good reason to have an exception for if there is an id.   Maybe there is??

**Comment by swirtSJW** [comment]
I like the looks of this button as opposed to the csv download link at the bottom, when there is an attached View.  However the attached View picks up the filters from the View.  This button does not. It provides the whole thing, unfiltered.
image

**Comment by msbtterswrth** [comment]
@swirtSJW I removed the nicer button in favor of the core attachment. I'll add a new ticket to explore how we can make that default csv button better for all views.

**Review by swirtSJW — APPROVED** [review]

**Pull request merged**